### PR TITLE
Replace hand-rolled TablelessModel with ActiveModel::Model

### PR DESCRIPTION
**Pull request by benilovj**
Since the move to Rails 4.0, it's possible to use the `ActiveModel::Model`
for models that aren't backed by a database. This change removes the hand-rolled
implementation.

More info here:
http://blog.plataformatec.com.br/2012/03/barebone-models-to-use-with-actionpack-in-rails-4-0/

### CODE REVIEW

File: spec/controllers/create_or_change_user_requests_controller_spec.rb [hunk 1]
```diff
@@ -28,7 +28,6 @@ def valid_change_user_request_params
           "email"=>"subject@digital.cabinet-office.gov.uk",
         },
         "action" => "change_user",
-        "tool_role" => "govt_form",
```
Comment (evilstreak) [inline spec/controllers/create_or_change_user_requests_controller_spec.rb]: This change doesn't seem to match the rest of this commit. Is it intentional?

Comment (benilovj) [inline spec/controllers/create_or_change_user_requests_controller_spec.rb]: Yes, it's intentional but I'll put it in a separate commit and add an appropriate comment

Comment (benilovj) [inline spec/controllers/create_or_change_user_requests_controller_spec.rb]: @evilstreak I've now split this change out into d838255. The reason why it's necessary to remove that redundant field is that `ActiveModel::Model` throws a wobbly if trying to set a non-existent field.

Comment (evilstreak) [inline spec/controllers/create_or_change_user_requests_controller_spec.rb]: Thanks for clarifying

**Comment by evilstreak** [comment]
:+1:

**Pull request merged**